app/services/profanity_detector.rb

```diff
@@ -49,15 +40,17 @@ def preprocess_if_needed!
     end
 
     # Map of {Integer => Regexp}
-    @regex_by_length = Hash.new
+    regex_by_length = Hash.new
 
     profanity_by_length.each do |k, v|
       escaped = v.to_a.map { |x| Regexp.escape(x) }
-      @regex_by_length[k] = Regexp.new("(#{escaped.join('|')})")
+      regex_by_length[k] = Regexp.new("(#{escaped.join('|')})")
     end
 
-    @min_profanity_length, @max_profanity_length = profanity_by_length.keys.minmax
+    min_profanity_length, max_profanity_length = profanity_by_length.keys.minmax
 
-    @preprocessed = true
+    return [regex_by_length, min_profanity_length, max_profanity_length]
```

Comment:
redundant `return` keyword
```suggestion
    [regex_by_length, min_profanity_length, max_profanity_length]
```